src/realm/views.cpp

```diff
@@ -190,9 +223,123 @@ bool SortDescriptor::Sorter::operator()(IndexPair i, IndexPair j, bool total_ord
     return total_ordering ? i.index_in_view < j.index_in_view : 0;
 }
 
-void RowIndexes::do_sort(const SortDescriptor& order, const SortDescriptor& distinct)
+DescriptorOrdering::DescriptorOrdering(const DescriptorOrdering& other)
+{
+    for (const auto& d : other.m_descriptors) {
+        m_descriptors.emplace_back(d->clone());
+    }
+}
+
+DescriptorOrdering& DescriptorOrdering::operator=(const DescriptorOrdering& rhs)
+{
+    m_descriptors.clear();
+    for (const auto& d : rhs.m_descriptors) {
+        m_descriptors.emplace_back(d->clone());
+    }
+    return *this;
+}
+
+void DescriptorOrdering::append_sort(SortDescriptor sort)
```

Comment:
I think this would be a little clearer if it were expressed like so:
```c++
void DescriptorOrdering::append_sort(SortDescriptor sort)
{
    if (!sort.is_valid()) {
        return;
    }

    if (!m_descriptors.empty()) {
        if (SortDescriptor* previous_sort = dynamic_cast<SortDescriptor*>(m_descriptors.back().get())) {
            previous_sort->merge_with(std::move(sort));
            return;
        }
    }
    m_descriptors.emplace_back(new SortDescriptor(std::move(sort)));
}
```